### PR TITLE
[WIP] Add keyboard shortcuts to navigate the client UI

### PR DESCRIPTION
**Pull request by tcbyrd**
I was working with the web UI a lot recently and found myself wishing there were a quick way to navigate to different deliveries with my keyboard so they could be redelivered or pinned easily. This is still a WIP, but I since there are a few different ways to tackle this, I thought I'd open this up to see how y'all like this approach. It uses [`react-hotkeys`](https://github.com/greena13/react-hotkeys), which is a React implementation built on top of [Mousetrap](https://craig.is/killing/mice).

This adds a higher-order component ([HOC](https://reactjs.org/docs/higher-order-components.html)) for the hotkeys, so since it needs to only wrap the ListItems, I renamed `App.js` to `Deliveries.js` and wrapped `<Deliveries />` in the HotKeys component inside `<App />`. It defaults to focusing on the list when it loads to ensure the hotkeys are usable immediately.

I'm partial to vim/gmail style hot keys, but this makes it fairly straightforward to map other keys to the handlers. Here's what's currently mapped:

```js
const keyMap = {
  moveDown: 'j', // Move to the next event
  moveUp: 'k', // Move to the previous event
  open: 'o', // Expand or "open" the selected event
  pin: 'p', // Pin the selected event
  redeliver: 'r' // Redeliver the selected event
}
```

TODO:
- [ ] Write handlers for the events
- [ ] Figure out how to write tests for keydown events and focus management
- [ ] Move App.test.js to Deliveries.test.js

### CODE REVIEW

File: server/src/components/Deliveries.js [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+          let [searchString, value] = filter.split(':')
+          if (!searchString.startsWith('body')) searchString = `body.${searchString}`
+          console.log(l, searchString, value)
+          return get(l, searchString) === value
+        }
+        return true
+      })
+    }
+    const handlers = {
+      'moveUp': (e) => console.log('pressed up'),
```
Comment (JasonEtco) [inline server/src/components/Deliveries.js]: I realize you're still working on the handlers, but these object keys don't need the single-quotes

**Comment by JasonEtco** [comment]
This is awesome - I think we should find a way to surface the available hotkeys somewhere in the UI. I don't know where, but I'll let you know if I find a good spot 👍

File: server/src/components/Deliveries.js (hunk 1 above)
Comment (tcbyrd) [reply server/src/components/Deliveries.js]: Yeh this is how they had it in their documentation. Seemed weird, but it worked. Will adjust

**Comment by tcbyrd** [comment]
I was thinking a little keyboard icon in the bottom right that pops up a modal with all the shortcuts. That modal should also have a hotkey (`shift + ?` to be consistent with gmail/github)

**Comment by tcbyrd** [comment]
Haven't had time to work on this, but would still like to get it finished along with some other improvements. Leaving this one open.

**Comment by tcbyrd** [comment]
Going to close this since I haven't made any real progress and we're splitting the repos. This may be a good excuse to play around with React Hooks later, though, so I may come back to it. Happy to review other PRs if anyone wants to give it a try.

**Pull request closed without merging**